homeassistant/components/tradfri/__init__.py

```diff
@@ -157,6 +160,9 @@ async def on_hass_stop(event: Event) -> None:
         )
         await coordinator.async_config_entry_first_refresh()
 
+        entry.async_on_unload(
+            async_dispatcher_connect(hass, SIGNAL_GW, coordinator.set_hub_available)
```

Comment:
Should we send a signal to the group coordinators too, that the gateway is unavailable?

Reply:
I was just thinking in those terms as well. But is that not over engineering it given that groups is an opt-in and has been for the past three (?) years?

Reply:
As long as we keep them around I think we should make the experience consistent for the different entities as far as possible. It would be weird if the groups stayed available when the other entities turned unavailable and the gateway is unavailable.

Reply:
It's a fair point, I'll add it.